### PR TITLE
feat(web/applications): ui table server-side (boas-594)

### PR DESCRIPTION
**Pull request by lutamKC**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/src/client/components/files-list/files-list.module.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	document.location = `.?number=1&size=${changeEvent.target.value}`;
+};
+
+/**
+ * Update the counter of the selected pages
+ *
+ */
+const updateSelectedFilesCounter = () => {
+	/** @type {HTMLElement | null} */
+	const selectedFilesNumber = document.querySelector('#selectedFilesNumber');
```
Comment (francescobrenci-AR) [inline apps/web/src/client/components/files-list/files-list.module.js:45]: get this DOM element with the others at the top of the file

File: apps/web/src/server/applications/pages/case/documentation/applications-documentation.controller.js [hunk 2]
```diff
@@ -7,7 +7,9 @@ import {
 /** @typedef {import('../applications-case.locals.js').ApplicationCaseLocals} ApplicationCaseLocals */
 /** @typedef {import('../../../applications.types').DocumentationCategory} DocumentationCategory */
 /** @typedef {import('./applications-documentation.types').CaseDocumentationUploadProps} CaseDocumentationUploadProps */
-/** @typedef {import('./applications-documentation.types').DocumentationPageProps} DocumentationPageProps */
+/** @typedef {import('./applications-documentation.types').CaseDocumentationBody} CaseDocumentationBody */
```
Comment (francescobrenci-AR) [inline apps/web/src/server/applications/pages/case/documentation/applications-documentation.controller.js:10]: there are some @ts errorts

File: apps/web/src/server/applications/pages/case/documentation/applications-documentation.controller.js [hunk 3]
```diff
@@ -25,27 +27,65 @@ export async function viewApplicationsCaseDocumentationCategories(request, respo
...
 }
 
+/**
+ * Change properties for the documentation files
+ *
+ *
+ * @type {import('@pins/express').RenderHandler<{}, {}, CaseDocumentationBody, {}, {}>}
+ */
+export async function updateApplicationsCaseDocumentationFolder(request, response) {
+	// console.log(request.body.selectedFilesIds);
```
Comment (francescobrenci-AR) [inline apps/web/src/server/applications/pages/case/documentation/applications-documentation.controller.js:85]: remove log

File: apps/web/src/server/views/applications/case-documentation/documentation-folder.njk [hunk 4]
```diff
@@ -1,67 +1,174 @@
...
+				<p>
+					<span id="selectedFilesNumber">0</span> document(s) selected
+				</p>
 			</div>
-		{% endfor %}
-	{% else %}
-		This folder contains no documents
-	{% endif %}
 
+			<table class="govuk-table pins-files-list--table pins-table">
```
Comment (francescobrenci-AR) [inline apps/web/src/server/views/applications/case-documentation/documentation-folder.njk:101]: Markup looks ok - worth checking how the table reads on Screen Readers

File: apps/web/src/styles/7-components/files-list.scss [hunk 5]
```diff
@@ -0,0 +1,95 @@
+@use "../govuk-frontend/govuk/settings/_colours-applied" as colors;
+
+.pins-files-list {
+  &--statuses, &--actions {
```
Comment (francescobrenci-AR) [inline apps/web/src/styles/7-components/files-list.scss:4]: `--` is for modifiers, in this case use `__`

File: apps/web/src/server/applications/pages/case/documentation/applications-documentation.service.js [hunk 6]
```diff
@@ -50,19 +52,30 @@ export const getCaseDocumentationFolderPath = (caseId, folderId) => {
...
+export const getCaseDocumentationFilesInFolder = (
+	caseId,
+	folderId,
+	pageSize = 50,
+	pageNumber = 0
+) => {
 	// TODO: Mock Version - to be replaced when API to get documents exists:
 	/** @type {DocumentationFile[] } */
-	const documentationFiles = [];
+	const documentationFiles = fixtureDocumentationFiles;
```
Comment (francescobrenci-AR) [inline apps/web/src/server/applications/pages/case/documentation/applications-documentation.service.js:67]: worth updating the tests to check for correct number of files

**Review by francescobrenci-AR — CHANGES_REQUESTED** [review]

File: apps/web/src/server/applications/pages/case/documentation/applications-documentation.controller.js (hunk 2 above)
Comment (lutamKC) [reply apps/web/src/server/applications/pages/case/documentation/applications-documentation.controller.js:14]: can't see them

File: apps/web/src/server/applications/pages/case/documentation/applications-documentation.service.js (hunk 6 above)
Comment (lutamKC) [reply apps/web/src/server/applications/pages/case/documentation/applications-documentation.service.js:67]: this is the mocked service (should be replaced by the api). the test for the correct number is in the controller

File: apps/web/src/server/applications/pages/case/documentation/applications-documentation.controller.js (hunk 2 above)
Comment (francescobrenci-AR) [reply apps/web/src/server/applications/pages/case/documentation/applications-documentation.controller.js:14]: strange - can see them my end

**Review by francescobrenci-AR — APPROVED** [review]

**Pull request merged**